src/Raven.Server/Indexing/LuceneVoronDirectory.cs

```diff
@@ -14,6 +14,7 @@ public sealed unsafe class LuceneVoronDirectory : Lucene.Net.Store.Directory
         private readonly StorageEnvironment _environment;
         private readonly string _name;
         private readonly IndexOutputFilesSummary _indexOutputFilesSummary;
+        private readonly TempFileCache _tmpFileCache;
```

Comment:
The problem that I can see here is that size of this can be 32 * 128kb = 4mb

Now if we will take that 4MB and multiply that by the amount of indexes then we are getting quite a lot of memory. Are we OK with that?
Do we want to subscribe to low memory notifications and clear the cache then?

Reply:
Good point, I'll add handling to that.